core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -0,0 +1,315 @@
+    Table table = catalog.buildTable(ident, request.schema())
+        .withLocation(request.location())
+        .withPartitionSpec(request.spec())
+        .withSortOrder(request.writeOrder())
+        .withProperties(request.properties())
+        .create();
+
+    if (table instanceof BaseTable) {
+      return LoadTableResponse.builder()
+          .withTableMetadata(((BaseTable) table).operations().current())
+          .build();
+    }
+
+    throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+  }
+
+  public static DropTableResponse dropTable(Catalog catalog, TableIdentifier ident) {
+    boolean dropped = catalog.dropTable(ident);
+    return DropTableResponse.builder().dropped(dropped).build();
+  }
+
+  public static LoadTableResponse loadTable(Catalog catalog, TableIdentifier ident) {
+    Table table = catalog.loadTable(ident);
+
+    if (table instanceof BaseTable) {
+      return LoadTableResponse.builder()
+          .withTableMetadata(((BaseTable) table).operations().current())
+          .build();
+    }
+
+    throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+  }
+
+  public static LoadTableResponse updateTable(Catalog catalog, TableIdentifier ident, UpdateTableRequest request) {
+    TableMetadata finalMetadata;
+    if (isCreate(request)) {
+      // this is a hacky way to get TableOperations for an uncommitted table
+      Transaction transaction = catalog.buildTable(ident, EMPTY_SCHEMA).createOrReplaceTransaction();
+      if (transaction instanceof BaseTransaction) {
+        BaseTransaction baseTransaction = (BaseTransaction) transaction;
+        finalMetadata = create(baseTransaction.underyingOps(), baseTransaction.startMetadata(), request);
+      } else {
+        throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTransaction");
+      }
+
+    } else {
+      Table table = catalog.loadTable(ident);
+      if (table instanceof BaseTable) {
+        TableOperations ops = ((BaseTable) table).operations();
+        finalMetadata = commit(ops, request);
+      } else {
+        throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+      }
+    }
+
+    return LoadTableResponse.builder()
+        .withTableMetadata(finalMetadata)
+        .build();
+  }
+
+  private static boolean isCreate(UpdateTableRequest request) {
+    boolean isCreate = request.requirements().stream()
+        .anyMatch(UpdateTableRequest.UpdateRequirement.AssertTableDoesNotExist.class::isInstance);
+
+    if (isCreate) {
+      List<UpdateTableRequest.UpdateRequirement> invalidRequirements = request.requirements().stream()
+          .filter(req -> !(req instanceof UpdateTableRequest.UpdateRequirement.AssertTableDoesNotExist))
+          .collect(Collectors.toList());
+      Preconditions.checkArgument(invalidRequirements.isEmpty(),
+          "Invalid create requirements: %s", invalidRequirements);
```

Comment:
So either all requirements assert that the table does not exist, or none do? If only some do we have an invalid Create request?

Reply:
Yes, that's correct.

If you're asserting that the table does not exist, then it makes no sense to assert that it has a certain current schema ID, for example.